traffic_server/plugins/astats_over_http/astats_over_http.c

```diff
@@ -298,19 +299,18 @@ static int getSpeed(char *inf, char *buffer, int bufferSize) {
 
 static void appendSystemState(stats_state *my_state) {
 	char *interface = my_state->interfaceName;
-	char buffer[2024];
-	int bsize = 2024;
+	char buffer[16384];
```

Comment:
What's the motivation for this? What problems were the old buffer size causing?

Reply:
```
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
br-5e66a6784dcd:       0       0    0    0    0     0          0         0      928      12    0    0    0     0       0          0
   em3:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
br-09a0dde5b54a:  244822    8694    0    0    0     0          0         0    58191     542    0    0    0     0       0          0
virbr0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
br-4d9440de4eba:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
docker0: 77867415 1412703    0    0    0     0          0         0 11219225352 2039122    0    0    0     0       0          0
  p4p1: 17166817434005 2351818698    0    2    0     0          0  14184099 109452120652 879661558    0    0    0     0       0          0
br-cdfbca58ca33:       0       0    0    0    0     0          0         0     5276      60    0    0    0     0       0          0
br-3f46f2f9970a:       0       0    0    0    0     0          0         0     8478     179    0    0    0     0       0          0
virbr0-nic:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
br-12e7ed6b66f3: 2271911   21947    0    0    0     0          0         0 151217170   24506    0    0    0     0       0          0
   em1:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
    lo: 366481874375030 12180805622    0    0    0     0          0         0 366481874375030 12180805622    0    0    0     0       0          0
br-96a55ab3fc5d:  185022    3546    0    0    0     0          0         0 29023485    4014    0    0    0     0       0          0
 bond0: 34395814195367 4714290784    0    0    0     0          0  28368240 221007483586 1752861461    0   12    0     0       0          0
  p4p2: 17228996761982 2362472091    0    3    0     0          0  14184141 111555363182 873199905    0    0    0     0       0          0
   em4:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
br-c61aab48073a: 49841107   90471    0    0    0     0          0         0 31814390  135500    0    0    0     0       0          0
   em2:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
```

Reply:
I assume you mean because the interface you're looking for is far enough down in the file that it's outside the length of the buffer, e.g. `em2`.

looking at the plugin itself, I'm not convinced that simply increasing the buffer size actually fixes the problem; it merely mitigates it. The file should be read in in buffer-sized chunks until EOF, not just read in up to the buffer size and the rest discarded. For as long as we read the file improperly, it will be possible for `/proc/net/dev`s to exist that defy interface detection.

Reply:
Mitigation is fine.  The plugin is so full of holes regarding unchecked conditions that it needs a general review.